### PR TITLE
add new scenarios to add and commitCtx

### PR DESCRIPTION
**Pull request by acosferreira**

### CODE REVIEW

File: pkg/commits/commits_test.go [hunk 1]
```diff
@@ -195,3 +196,134 @@ func TestServeRepo(t *testing.T) {
...
+		got := response.Code
+		want := http.StatusOK
+		if got != want {
+			t.Errorf("got %q", got)
+		}
+	})
+}
+
+func TestAddError(t *testing.T) {
+	err := db.DB.AutoMigrate(models.Commit{})
```
Comment (biancarosa) [inline pkg/commits/commits_test.go:221]: Can we run automigrate in `setUp`?

File: pkg/commits/commits_test.go [hunk 2]
```diff
@@ -79,32 +80,23 @@ func TestGetAllEmpty(t *testing.T) {
...
 		request, _ := http.NewRequest(http.MethodGet, "/", nil)
 		response := httptest.NewRecorder()
 
 		GetAll(response, request)
-		got := response.Body.String()
-		if !strings.Contains(got, "0000000") {
+		got := response.Code
+		want := http.StatusOK
+		if got != want {
 			t.Errorf("got %q", got)
```
Comment (boaz0) [inline pkg/commits/commits_test.go:92]: Nitpick:
```go
t.Errorf("Expected status code to be %q but got %q", want, got)
```

File: pkg/commits/commits_test.go [hunk 3]
```diff
@@ -195,3 +168,114 @@ func TestServeRepo(t *testing.T) {
...
+		request, _ := http.NewRequest(http.MethodGet, "/", bytes.NewBuffer(jsonStr))
+		response := httptest.NewRecorder()
+		ctx := request.Context()
+		ctx = context.WithValue(ctx, commitKey, &cmt)
+		request = request.WithContext(ctx)
+		Patch(response, request)
+		got := response.Code
+		want := http.StatusBadRequest
+		if got != want {
+			t.Errorf("got %q", got)
```
Comment (boaz0) [inline pkg/commits/commits_test.go:257]: :waffle:

File: pkg/commits/commits_test.go [hunk 4]
```diff
@@ -195,3 +168,114 @@ func TestServeRepo(t *testing.T) {
...
+		request, _ := http.NewRequest(http.MethodGet, "/", bytes.NewBuffer(jsonStr))
+		response := httptest.NewRecorder()
+		ctx := request.Context()
+		ctx = context.WithValue(ctx, commitKey, &cmt)
+		request = request.WithContext(ctx)
+		Patch(response, request)
+		got := response.Code
+		want := http.StatusOK
+		if got != want {
+			t.Errorf("got %q", got)
```
Comment (boaz0) [inline pkg/commits/commits_test.go:237]: :sunrise_over_mountains:

File: pkg/commits/commits_test.go [hunk 5]
```diff
@@ -195,3 +168,114 @@ func TestServeRepo(t *testing.T) {
...
+		request, _ := http.NewRequest(http.MethodGet, "/", bytes.NewBuffer(jsonStr))
+		response := httptest.NewRecorder()
+		ctx := request.Context()
+		ctx = context.WithValue(ctx, commitKey, &cmt)
+		request = request.WithContext(ctx)
+		Update(response, request)
+		got := response.Code
+		want := http.StatusOK
+		if got != want {
+			t.Errorf("got %q", got)
```
Comment (boaz0) [inline pkg/commits/commits_test.go:218]: here 3 :smile:

File: pkg/commits/commits_test.go [hunk 6]
```diff
@@ -195,3 +168,114 @@ func TestServeRepo(t *testing.T) {
...
+
+		var jsonStr = []byte(`{bad json}`)
+		request, _ := http.NewRequest(http.MethodGet, "/", bytes.NewBuffer(jsonStr))
+		response := httptest.NewRecorder()
+
+		Add(response, request)
+		got := response.Code
+		want := http.StatusBadRequest
+		if got != want {
+			t.Errorf("got %q", got)
```
Comment (boaz0) [inline pkg/commits/commits_test.go:199]: here too :smiley:

File: pkg/commits/commits_test.go [hunk 7]
```diff
@@ -195,3 +168,114 @@ func TestServeRepo(t *testing.T) {
...
+
+		var jsonStr = []byte(`{ "Account": "123", "Name" :"test" }`)
+		request, _ := http.NewRequest(http.MethodGet, "/", bytes.NewBuffer(jsonStr))
+		response := httptest.NewRecorder()
+
+		Add(response, request)
+		got := response.Code
+		want := http.StatusOK
+		if got != want {
+			t.Errorf("got %q", got)
```
Comment (boaz0) [inline pkg/commits/commits_test.go:183]: same as above:

```go
t.Errorf("Expected status code to be %q but got %q", want, got)
```

File: pkg/commits/commits_test.go [hunk 8]
```diff
@@ -195,3 +168,114 @@ func TestServeRepo(t *testing.T) {
...
+	// if err != nil {
+	// 	panic(err)
+	// }
+	mockCommit()
+
+	t.Run("returns Get commitCtx ", func(t *testing.T) {
+		next := http.HandlerFunc(final)
+		got := CommitCtx(next)
+		if got == nil {
+			t.Errorf("got %q", got)
```
Comment (boaz0) [inline pkg/commits/commits_test.go:274]: last one

File: pkg/commits/commits_test.go [hunk 9]
```diff
@@ -195,3 +168,114 @@ func TestServeRepo(t *testing.T) {
...
+		want := http.StatusBadRequest
+		if got != want {
+			t.Errorf("got %q", got)
+		}
+	})
+}
+
+//To continue
+func TestCommitCtx(t *testing.T) {
+	// err := db.DB.AutoMigrate(models.Commit{})
```
Comment (boaz0) [inline pkg/commits/commits_test.go:264]: should we remove these comments?

**Review by boaz0 — CHANGES_REQUESTED** [review]
Few small changes. But looks amazing! Thanks!

**Review by boaz0 — APPROVED** [review]

**Review by biancarosa — APPROVED** [review]

**Pull request merged**